jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResource.java

```diff
@@ -50,15 +50,17 @@ private Path checkAliasPath()
     {
         Path abs = path;
 
+        boolean isJarArchive = uri.toASCIIString().startsWith("jar:file:");
+
         /* Catch situation where the Path class has already normalized
          * the URI eg. input path "aa./foo.txt"
-         * from an #addPath(String) is normalized away during
+         * from an #resolve(String) is normalized away during
          * the creation of a Path object reference.
-         * If the URI is different then the Path.toUri() then
+         * If the URI is different from the Path.toUri() then
          * we will just use the original URI to construct the
          * alias reference Path.
          */
-        if (!URIUtil.equalsIgnoreEncodings(uri, path.toUri()))
+        if (!isJarArchive && !URIUtil.equalsIgnoreEncodings(uri, path.toUri()))
```

Comment:
I don't understand why jars are being excluded.  Please explain in the comment above.

Reply:
We have test cases for this now. See point 1 in comment above.